src/_includes/css/demos.css

```diff
@@ -0,0 +1,287 @@
+    :root {
+      --font: -apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, Oxygen, Ubuntu, Cantarell, "Open Sans", "Helvetica Neue", sans-serif;
+      --color-bg: black;
+      --color-fg: white;
+      --color-off: #333;
+      --color-accent: #fb0;
+      --border: 6px;
+      --spacer: 2rem;
+      --font-xl: 4rem;
+      --font-lg: 2.7rem;
+      --font-md: 2.2rem;
+      --font-sm: 1.1rem;
+      --image-width: 100%;
+    }
+
+    @media (min-width: 800px) {
+      :root {
+        --border: 8px;
+        --font-xl: 6rem;
+        --font-lg: 3.6rem;
+        --font-md: 2.8rem;
+        --font-sm: 1.6rem;
+        --image-width: 300px;
+        --spacer: 4rem;
+      }
+    }
+
+    @media (min-width: 1200px) {
+      :root {
+        --image-width: 380px;
+      }
+    }
+
+    @media (prefers-color-scheme: light) {
+      :root {
+        --color-bg: white;
+        --color-fg: black;
+        --color-off: #ebebeb;
+      }
+    }
+
+    * {
+      box-sizing: border-box;
+    }
+
+    @supports selector(A:focus-visible) {
+      :focus {
+        outline: none;
+      }
+    }
+
+    body,
+    html {
+      height: 100%;
+      margin: 0;
+    }
+
+    body {
+      background: var(--color-bg);
+      color: var(--color-fg);
+      font-family: var(--font);
+      font-weight: 300;
+      overflow-x: hidden;
+    }
+
+    h1,
+    h2 {
+      color: var(--color-fg);
+    }
+
+    h3 {
+      color: var(--color-off);
+    }
+
+    h2.hang {
+      text-indent: -0.5em;
+    }
+
+    h1,
+    h2,
+    h3,
+    p {
+      margin: 0;
+    }
+
+    h1 {
+      font-size: var(--font-xl);
+    }
+
+    h2 {
+      font-size: var(--font-lg);
+    }
+
+    h3 {
+      font-size: var(--font-md);
+    }
+
+    p {
+      font-size: var(--font-sm);
+    }
+
+    h3,
+    p {
+      margin-top: 0.5em;
+    }
+
+    h1,
+    h2,
+    h3 {
+      font-family: var(--font);
+      line-height: 1;
+    }
+
+    p {
+      line-height: 1.3;
+    }
+
+    a {
+      color: var(--color-fg);
+    }
+
+    h2 a {
+      color: inherit;
+      display: block;
+      text-decoration: none;
+    }
+
+    a:active,
+    a:focus-visible {
+      color: var(--color-accent);
+    }
+
+    @media (hover: hover) {
+      a:hover {
+        color: var(--color-accent);
+      }
+    }
+
+    main {
+      margin: 0 auto;
+      max-width: 1200px;
+      padding-bottom: calc(var(--spacer) * 2);
+      width: calc(100% - var(--spacer) * 2);
+    }
+
+    main header {
+      padding-top: calc(var(--spacer) * 2);
+      padding-bottom: calc(var(--spacer) * 0.5);
+      border-bottom: var(--border) solid var(--color-fg);
+    }
+
+    .filter-javascript nav li.javascript a,
```

Comment:
We need to **figure out a way to create these classes programmatically**, since we don't know which tags they'll need to be created for.

One idea is to:

1. **Create a JavaScript function** that iterates over the unique demo tags and produces a .json data file. 
2. This .json file could then be passed to a json-to-scss or json-to-css converter.
3. The final css/sass file should be included only for the Demos page (either in the layout or the demos/index.njk file).

Some useful links:

- https://github.com/11ty/eleventy/issues/1311#issuecomment-660719378
- https://www.11ty.dev/docs/data-js/
- https://github.com/11ty/eleventy/issues/1176
- https://piccalil.li/quick-tip/create-json-feed-eleventy/
- https://github.com/desirable-objects/json-to-css
- https://github.com/rlapoele/json-to-scss

These lines from `package.json` could be very useful: https://github.com/death-by-coconut/juancarlosasensio.com/blob/codepen-demos/package.json#L33-L34